Objects/typeobject.c

```diff
@@ -11127,6 +11136,18 @@ add_operators(PyTypeObject *type, PyTypeObject *def)
 }
 
 
+int
+PyType_Freeze(PyTypeObject *type)
+{
+    PyObject *bases = type->tp_bases;
+    if (check_immutable_bases(type->tp_name, bases) < 0) {
```

Comment:
I don't understand well the difference between MRO and Bases here. Do you have an example where checking MRO makes a difference?

Reply:
I don't have an example right now, but, a metaclass could override [`mro()`](https://docs.python.org/3.14/library/stdtypes.html#class.mro) to add a mutable “base”.

Reply:
I added a more elaborate test with multiple inheritance: D inherits from A and C, and C inherits from B.

IMO checking base classes is enough, since a base class can only be immutable if its parent is also immutable. For example, C cannot be immutable if B is mutable.

Changing MRO in a metaclass is really complicated. If you do that, you're playing with fire, and and you're on your own: https://stackoverflow.com/questions/20822850/change-python-mro-at-runtime

Reply:
Complicated or not, it should not be allowed to invalidate CPython's invariants.

Reply:
I modified my PR to check ``__mro__`` instead of ``__bases__``.